specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,243 @@
+according by the `orderBy` parameter: either by the most recent start times, the most recent
+end times, or the longest time spent on the activity.
+
+Each parameter in `queries` is ORed together in the resulting database query, whereas the contents
+of a single object in the `queries` array are ANDed together.  For example, if you provided two
+queries, each of which contained the keywords "tax", one of which had a content type of
+"application/pdf" and the other had a content type of "image/*", the resulting database query
+would be something along these lines:
+
+```sql
+SELECT * FROM UserActivityHistory WHERE
+    (CONTAINS (DisplayText, '"tax"')) AND
+    ((CONTAINS (ContentType, '"application/pdf"')) OR
+     (CONTAINS (ContentType, '"image/*"')));
+```
+
+## UserActivityHistory.SearchAsync method
+
+This method is an asynchronous version of the `Search` method.
+
+## UserActivityHistory.GetAppsWithUserActivity method
+
+This method synchronously retrieves a list of all the app names with data in the user's activity
+history database. You can use this, for example, to show the user the list of apps that are being
+queried against, so the user can understand why an app that is not recording user activity is not
+showing up in the results.
+
+## UserActivityHistory.GetAppsWithUserActivityAsync method
+
+This method is an asynchronous version of the `GetAppsWithUserActivity` method.
+
+## UserActivityHistoryItem class
+
+This class represents a single item in the user's activity history. It contains properties that
+describe in what app the activity occurred, what the nature of the activity was, the URI of the
+resource involved in the activity (e.g., a document, a webpage, a video, etc.), the URI that
+can be used to bring back the state the user left the activity in, and the times when the user
+started the activity and ended the activity.
+
+If the user performed the same activity multiple times, there will be multiple
+`UserActivityHistoryItem` objects returned, each with different start and end times.
+
+## UserActivityHistoryItem.AppName property
+
+This property contains the name of the app in which the activity occurred.
+
+## UserActivityHistoryItem.ActivityId property
+
+This property contains the ID of the activity, which can be used to collate multiple sessions
+of the same activity. For example, if the user watched a movie in multiple sessions, the
+`ActivityId` property can be used to identify how long in total the user spent watching that movie.
+
+## UserActivityHistoryItem.DisplayText property
+
+This property contains a string that is how the app chose to describe the activity. For example,
+if the activity was reading the contents of a webpage, this property might contain the webpage's title.
+
+## UserActivityHistoryItem.ContentType property
+
+This property contains the MIME type of the content being interacted with. For example, if the user
+was looking at a PNG image, this property would contain the string "image/png".
```

Comment:
is this be "anything the app gave Windows", or "one from a list of known MIME types"?
e.g. can Contoso say "contoso/bloop"?

Reply:
It's a string property, so it can be anything.  I can make that clear in the description.